File: scala/src/com/runtimeverification/ktest/kthings/bexp.scala
package com.runtimeverification.ktest.kthings

import com.runtimeverification.ktest.KThing

object bexp {
  class Bexp extends KThing
  case class Bool(value : KBool) extends Bexp
  case class LessOrEquals(left : aexp.Aexp, right : aexp.Aexp) extends Bexp
  case class Not(operand : Bexp) extends Bexp
  case class And(left : Bexp, right : Bexp) extends Bexp
}

File: java.optimized/src/main/java/com/runtimeverification/ktest/nonterminals/Bexp.java
package com.runtimeverification.ktest.nonterminals;

import com.runtimeverification.ktest.KThing;

public abstract class Bexp {
    public static KThing lessOrEquals(KThing first, KThing second) {
        return new KThing(KThing.Type.BEXP_LESS_OR_EQUALS, first, second);
    }

    public static KThing and(KThing first, KThing second) {
        return new KThing(KThing.Type.BEXP_AND, first, second);
    }

    public static KThing not(KThing operand) {
        return new KThing(KThing.Type.BEXP_NOT, operand);
    }

    public static KThing bool(KThing value) {
        return new KThing(KThing.Type.BEXP_BOOL, value);
    }
}
